src/test/scala/chiselTests/LazyCloneSpec.scala

```diff
@@ -57,6 +59,40 @@ class LazyCloneSpec extends ChiselFlatSpec {
       }))
     }
     ChiselStage.elaborate(new MyModule)
-    Counter.count should be(4L)
+    Counter.count should be(2L)
+  }
+
+  it should "clone Records because of external refs" in {
```

Comment:
Should we have a "not cloned" case for Record that is similar to the Bundle case above? Is there anything different about records and vecs in that way?

Reply:
I have added the "not cloned" case for Records.

Good point on Vecs, they don't have special behavior here but definitely have the same issue with needing to recursively check Bundles inside of them for external refs... I want to fix that in a separate PR though to avoid scope creep on this one.

Reply:
Actually I don't think Vecs need to check recursively because they always clone their gen argument, but I will add tests to illustrate this behavior!